pe.c

```diff
@@ -873,6 +873,63 @@ handle_sbat(char *SBATBase, size_t SBATSize)
 	return efi_status;
 }
 
+static EFI_STATUS verify_image(void *data, unsigned int datasize,
+			       EFI_LOADED_IMAGE *li,
+			       PE_COFF_LOADER_IMAGE_CONTEXT *context)
+{
+	EFI_STATUS efi_status;
+	UINT8 sha1hash[SHA1_DIGEST_SIZE];
+	UINT8 sha256hash[SHA256_DIGEST_SIZE];
+
+	/*
+	 * The binary header contains relevant context and section pointers
+	 */
+	efi_status = read_header(data, datasize, context);
+	if (EFI_ERROR(efi_status)) {
+		perror(L"Failed to read header: %r\n", efi_status);
+		return efi_status;
+	}
+
+	/*
+	 * We only need to verify the binary if we're in secure mode
+	 */
+	efi_status = generate_hash(data, datasize, context, sha256hash,
+				   sha1hash);
+	if (EFI_ERROR(efi_status))
+		return efi_status;
+
+	/* Measure the binary into the TPM */
+#ifdef REQUIRE_TPM
+	efi_status =
+#endif
+	tpm_log_pe((EFI_PHYSICAL_ADDRESS)(UINTN)data, datasize,
+		   (EFI_PHYSICAL_ADDRESS)(UINTN)context->ImageAddress,
+		   li->FilePath, sha1hash, 4);
+#ifdef REQUIRE_TPM
+	if (efi_status != EFI_SUCCESS) {
+		return efi_status;
+	}
+#endif
```

Comment:
I think it might be cleaner to do something like:

```C
    efi_status = tpm_log_pe(...);
#ifdef REQUIRE_TPM
    if (efi_status != EFI_SUCCESS) {
        return efi_status;
    }
#endif
    (void)efi_status;
```

Also, should the check be `EFI_ERROR(efi_status)` here?

Reply:
The history behind this change is here:  https://github.com/rhboot/shim/commit/d737c0273d933f7348eb7adafd4ae772adc9c1b0